### PR TITLE
Feat/oauth migration

### PR DESCRIPTION
**Pull request by 0xtiki**

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -66,7 +66,7 @@ const HeroButton = () => {
     <Button
       color='gray.700'
       leftIcon={<FaSignInAlt />}
-      onClick={() => signIn('discord', { callbackUrl: '/dashboard' })}>
+      onClick={() => ''}>
```
Comment (wpmonty) [inline pages/index.tsx:71]: This should probably do something

File: pages/api/[...slug].ts [hunk 2]
```diff
@@ -11,8 +11,13 @@ export default async function handler(
...
     const url = typeof(req.query.slug)==='string' ? req.query.slug : req.query.slug.join('/')
 
+    console.log( url)
+
+
     // trim trailing slash from api endpoint.
-    const apiEndPoint = `${process.env.GOVERNATOR_API_ENDPOINT?.replace(/\/$/, '')}`
+    const apiEndPoint = `${process.env.NEXT_PUBLIC_GOVERNATOR_API_ENDPOINT?.replace(/\/$/, '')}`
+
+    console.log(`${apiEndPoint}/${url}`)
```
Comment (wpmonty) [inline pages/api/[...slug].ts]: logs

File: hooks/useServers.ts [hunk 3]
```diff
@@ -48,9 +46,10 @@ const useServers = () => {
         if (retry < 4) {
           setTimeout(fetchData, 500);
           setRetry(retry + 1)
-        } else {
-          signOut()
-        }
+        } 
+        // else {
+        //   ((signOut()))
```
Comment (wpmonty) [inline hooks/useServers.ts]: is this signout logic no longer useful?

File: components/RouteGuard.tsx [hunk 4]
```diff
@@ -20,13 +20,13 @@ const RouteGuard: React.FC = ({ children }) => {
   const isAllowed = allowedPages.includes(router.route)
 
   useEffect(() => {
-    if (status === 'unauthenticated' && !isAllowed) {
+    if ((!session || session.status === 401) && !isAllowed) {
```
Comment (wpmonty) [inline components/RouteGuard.tsx]: maybe should be `|| !isAllowed`

File: components/NavBar.tsx [hunk 5]
```diff
@@ -173,24 +181,26 @@ const MobileDrawer: React.FC<{ session: Session | null }> = ({ session }) => {
...
                   >Feedback</Button>
                 </chakra.a>
-                <Button
-                  onClick={() => {
-                    signOut()
-                  }}>
-                  Logout
-                </Button>
+                <Text
+                  // onClick={() => {
```
Comment (wpmonty) [inline components/NavBar.tsx]: rm

File: components/NavBar.tsx [hunk 6]
```diff
@@ -46,24 +46,32 @@ const LoginText = () => {
...
   const { isOpen, onOpen, onClose } = useDisclosure()
   const btnRef = useRef(null);
   const router = useRouter();
 
-  const name = session?.user?.name
-  const image = session?.user?.image
-  
+  let name = '';
+  let image = '';
+  const loginUrl = `${process.env.NEXT_PUBLIC_GOVERNATOR_API_ENDPOINT}/auth/login`;
```
Comment (wpmonty) [inline components/NavBar.tsx]: so.. user with navigate to API? will need to test this in vercel

File: components/NavBar.tsx [hunk 7]
```diff
@@ -301,7 +311,9 @@ const NavBar = () => {
     checkAndCreateUser().then(() => null);
 
     //eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [session?.discordId])
+  }, [session])
+
+  const loginUrl = `${process.env.NEXT_PUBLIC_GOVERNATOR_API_ENDPOINT}/auth/login`;
```
Comment (wpmonty) [inline components/NavBar.tsx]: `loginUrl` is defined above? Maybe can be moved to constants file

File: pages/index.tsx (hunk 1 above)
Comment (0xtiki) [reply pages/index.tsx:71]: nah, not needed anymore. I tried to delete it but the button won't work without it.

File: hooks/useServers.ts (hunk 3 above)
Comment (0xtiki) [reply hooks/useServers.ts]: not needed anymore

File: components/RouteGuard.tsx (hunk 4 above)
Comment (0xtiki) [reply components/RouteGuard.tsx]: it's correct. If there is no session or the session is unauthenticated AND route is not public

File: components/NavBar.tsx (hunk 6 above)
Comment (0xtiki) [reply components/NavBar.tsx]: yes. Hope it works. You need to rename the env variable to make it public

File: pages/api/[...slug].ts [hunk 8]
```diff
@@ -11,8 +11,13 @@ export default async function handler(
 
     const url = typeof(req.query.slug)==='string' ? req.query.slug : req.query.slug.join('/')
 
+    console.log( url)
+
+
```
Comment (zberwaldt) [inline pages/api/[...slug].ts]:
```suggestion

```

File: pages/api/[...slug].ts (hunk 2 above)
Comment (zberwaldt) [inline pages/api/[...slug].ts]:
```suggestion
const apiEndPoint = `${process.env.NEXT_PUBLIC_GOVERNATOR_API_ENDPOINT?.replace(/\/$/, '')}`
```

Comment (zberwaldt) [reply pages/api/[...slug].ts]: @tikisailor Added suggestions to easily remove logs

File: pages/index.tsx (hunk 1 above)
Comment (zberwaldt) [reply pages/index.tsx:71]: I'll take a look at this when I have a chance

**Review by zberwaldt — APPROVED** [review]
Overall looks goot to me.

File: components/NavBar.tsx (hunk 6 above)
Comment (wpmonty) [reply components/NavBar.tsx]: so yeah FE is just navigating to the API, which just is a JSON page. Not useable. Maybe something is wrong? https://governator-staging-h8cdb8ie3-governator.vercel.app/

Comment (0xtiki) [reply components/NavBar.tsx]: it's a redirect to the discord oauth, has to come from same site that is why we redirect to backnd first. This part works. What doesn't work is sending the session cookie cross site. Lets see if we can get BE and FE under the same domain, that should solve the cookie and also we could implement this part with an axios request probably.

**Comment by wpmonty** [comment]
I'm able to login! But, I would like to be redirected to the Dashboard when login is complete. Currently, it returns me to homepage after auth.

Channels and Roles are missing. I also don't see any Polls in my servers.

**Review by wpmonty — APPROVED** [review]

**Pull request merged**